packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -312,40 +339,48 @@ export class ProjectDataSource {
     this._specWatcher.on('all', onProjectFileSystemChange)
   }
 
-  async defaultSpecFileName () {
-    const getDefaultFileName = (testingType: TestingType) => `cypress/${testingType}/${testingType === 'e2e' ? 'spec' : 'ComponentName'}.cy.${this.ctx.lifecycleManager.fileExtensionToUse}`
+  async defaultSpecFileName (): Promise<string | null> {
+    const defaultFilename = `${this.ctx.coreData.currentTestingType === 'e2e' ? 'spec' : 'ComponentName'}.cy.${this.ctx.lifecycleManager.fileExtensionToUse}`
+    const defaultPathname = path.join('cypress', this.ctx.coreData.currentTestingType ?? 'e2e', defaultFilename)
 
     try {
       if (!this.ctx.currentProject || !this.ctx.coreData.currentTestingType) {
-        return getDefaultFileName('e2e')
+        return null
```

Comment:
defaultSpecFileName is nonNull in the schema now. I think throwing an error here would be appropriate, if we don't want to just return the e2e default like below.

Reply:
Or we can change `defaultSpecFileName` to be nullable in the schema, either way

Reply:
This was recently changed to be nonNull to remove some of the guards/duplicated default logic that was present in app. Agreed it's not a big deal either way tho, we could default to an empty string in the app and have a similar result.

Reply:
Whoops, fixed